diff --git a/source/microbit/main.cpp b/source/microbit/main.cpp
index f30db1a..edfe9d5 100644
--- a/source/microbit/main.cpp
+++ b/source/microbit/main.cpp
@@ -3,7 +3,7 @@
 #include "microbit/memory.h"
 #include "microbit/filesystem.h"
 #include "microbit/microbitdal.h"
-#include "_newlib_version.h"
+//#include "_newlib_version.h"
 
 // Global instances of the DAL components that we use
 MicroBitDisplay ubit_display;
diff --git a/source/microbit/microbitcompass.cpp b/source/microbit/microbitcompass.cpp
index 062dbe8..fc1a7e8 100644
--- a/source/microbit/microbitcompass.cpp
+++ b/source/microbit/microbitcompass.cpp
@@ -44,7 +44,7 @@ void microbit_compass_init(void) {
     // load any peristent calibration data if it exists
     uint32_t *persist = (uint32_t*)microbit_compass_calibration_page();
     if (persist[0] == COMPASS_CALIBRATION_MAGIC) {
-        CompassSample samp = {(int)persist[1], (int)persist[2], (int)persist[3]};
+        CompassSample samp((int)persist[1], (int)persist[2], (int)persist[3]);
         ubit_compass.setCalibration(samp);
     }
 }
diff --git a/source/microbit/microbitpinmode.c b/source/microbit/microbitpinmode.c
index d3a1c6c..d9a7a42 100644
--- a/source/microbit/microbitpinmode.c
+++ b/source/microbit/microbitpinmode.c
@@ -93,7 +93,7 @@ const microbit_pinmode_t microbit_pinmodes[] = {
     [MODE_WRITE_ANALOG]  = { MP_QSTR_write_analog, analog_release },
     [MODE_READ_DIGITAL]  = { MP_QSTR_read_digital, noop },
     [MODE_WRITE_DIGITAL] = { MP_QSTR_write_digital, noop },
-    [MODE_DISPLAY]       = { MP_QSTR_display, pinmode_error },
+    [MODE_DISPLAY]       = { MP_QSTR_display, analog_release },
     [MODE_BUTTON]        = { MP_QSTR_button, pinmode_error },
     [MODE_MUSIC]         = { MP_QSTR_music, pinmode_error },
     [MODE_AUDIO_PLAY]    = { MP_QSTR_audio, noop },
